### DISCUSSION TITLE
revalidateTag  for non fetch functions like prisma

### DISCUSSION THREAD

**Discussion by alsherif-khalaf** [vercel/next.js#52854] [App Router]
How can I tag this function for example :

```ts
  const todos = await prisma.todo.findMany();
```
like this :

```ts
fetch(url, { next: { tags: [...] } });
```

to use it via revalidateTag('tag') ?

**Comment by mathiasrscom** [comment]
Hi [alsherif-khalaf](https://github.com/alsherif-khalaf)

The only way I could get it to work was by using route handlers, and then I added the prisma logic in there. 

Here is an example where I'm using fetch to an API route: 

```js
async function getProfileByUserId(id: string ) {
  const url = `${basePath}/api/read/profile/getProfileByUserId?id=${id}`

  const res = await fetch(url, { next: { tags: ["user"] } })

  return res.json()
}
```

If you found a better way, please let me know.

**Comment by dsosborn** [comment]
Check out `unstable_cache`. https://nextjs.org/docs/app/api-reference/functions/unstable_cache
I think this is a perfect use case for it, caching non-fetch data requests.

**Reply by tarek-mo** [reply]
Can i revalidate the cache using `revalidateTag` when I use this unstable_cache function?
